push_reference_update.py

```diff
@@ -0,0 +1,128 @@
+"""
+Script file for generating reference updates
+"""
+
+from idseq_pipeline.commands.common import *
+
+local_work_dir = "idseq_pipeline_temp"
+ncbitool_s3_path = "s3://czbiohub-infectious-disease/ncbitool"
+
+
+def main():
+    ##### PARAMETERS TO EDIT #####
+
+    ## To pull NCBI references through ncbitool, which archives and records versions, set URL_PREFIX=''.
+    # To pull the latest files from NCBI without using ncbitool (no version information), set URL_PREFIX=ftp://ftp.ncbi.nlm.nih.gov.
+    os.environ["URL_PREFIX"] = "ftp://ftp.ncbi.nlm.nih.gov"
+
+    ## Instances with adequate resources for making gsnap and rapsearch indexes.
+    # Need to have write access to destination folders.
+    os.environ["RAPSEARCH_SERVER_IP"] = "54.191.193.210"
+    os.environ["GSNAP_SERVER_IP"] = "34.211.67.166"
+
+    ##### COMMANDS #####
+    date_setup()
+    make_gsnap_index()
+    make_rapsearch_index()
+    make_lineage_files()
+    make_accession_mapping()
+
+
+def make_gsnap_index():
+    nt = "/blast/db/FASTA/nt.gz"
+    date = set_index_date([nt])
```

Comment:
do we want to set_index_date differently for each of the part of our index. I feel it would be easier to take a snapshot for all the files we need on the same time and use the start of running time for index making

Reply:
@yunfangjuan @cdebourcy Ah right so I was thinking to set "Run Time" as 'start of running time for gsnap+rapsearch+etc'. But do you think our "Source Time" should also be a single date across gsnap+rapsearch+etc? Or by snapshot did you mean actually copying the files again?

Reply:
By snapshoting, I mean the time you get the files. So I think all of them can share the same source time at which we snapshot the data (which is slightly different from when the data was actaully updated at NCBI side.)